File: 674/a.cpp

#include <bits/stdc++.h>

typedef long long ll;

std::map <ll, ll> f;

ll rec(ll x)
{
	return x < 4 ? x == 3 : (f.count(x) ? f[x] : f[x] = rec(x / 2) + rec((x + 1) / 2));
}

int main()
{
	ll n;
	std::cin >> n;
	std::cout << rec(n);
}

File: 674/a.py
f = {0:0, 1:0, 2:0, 3:1}
def rec(x):
	if x not in f: f[x] = rec(x // 2) + rec((x + 1) // 2)
	return f[x];
n = int(input())
print(rec(n))